sam_view.c

```diff
@@ -560,211 +560,428 @@ static const char *copied_tags[] = { "RG", "BC", "QT", NULL };
+    } else {
+        return READ_UNKNOWN;
+    }
+}
+
+// Transform a bam1_t record into a string with the FASTQ representation of it
+// @returns true for error, false for success
+static bool bam1_to_fq(const bam1_t *b, kstring_t *linebuf, const bam2fq_state_t *state)
+{
+    int i;
+    int32_t qlen = b->core.l_qseq;
+    assert(qlen >= 0);
+    uint8_t *seq;
+    uint8_t *qual = bam_get_qual(b);
+    const uint8_t *oq = NULL;
+    if (state->use_oq) oq = bam_aux_get(b, "OQ") + 1;
+    bool has_qual = (qual[0] != 0xff || (state->use_oq && oq)); // test if there is quality
+
+    linebuf->l = 0;
+    // Write read name
+    readpart readpart = which_readpart(b);
+    kputc(state->filetype == FASTA? '>' : '@', linebuf);
+    kputs(bam_get_qname(b), linebuf);
+    // Add the /1 /2 if requested
+    if (state->has12) {
+        if (readpart == READ_1) kputs("/1", linebuf);
+        else if (readpart == READ_2) kputs("/2", linebuf);
+    }
+    if (state->copy_tags) {
+        for (i = 0; copied_tags[i]; ++i) {
+            uint8_t *s;
+            if ((s = bam_aux_get(b, copied_tags[i])) != 0) {
+                kputc('\t', linebuf);
+                kputsn(copied_tags[i], 2, linebuf);
+                kputsn(":Z:", 3, linebuf);
+                kputs(bam_aux2Z(s), linebuf);
+            }
+        }
+    }
+    kputc('\n', linebuf);
+
+    seq = bam_get_seq(b);
+
+    if (b->core.flag & BAM_FREVERSE) { // read is reverse complemented
+        for (i = qlen-1; i > -1; --i) {
+            char c = seq_nt16_str[seq_comp_table[bam_seqi(seq,i)]];
+            kputc(c, linebuf);
+        }
+    } else {
+        for (i = 0; i < qlen; ++i) {
+            char c = seq_nt16_str[bam_seqi(seq,i)];
+            kputc(c, linebuf);
+        }
+    }
+    kputc('\n', linebuf);
+
+    if (state->filetype == FASTQ) {
+        // Write quality
+        kputs("+\n", linebuf);
+        if (has_qual) {
+            if (state->use_oq && oq) {
+                if (b->core.flag & BAM_FREVERSE) { // read is reverse complemented
+                    for (i = qlen-1; i > -1; --i) {
+                        kputc(oq[i], linebuf);
+                    }
+                } else {
+                    kputs((char*)oq, linebuf);
+                }
+            } else {
+                if (b->core.flag & BAM_FREVERSE) { // read is reverse complemented
+                    for (i = qlen-1; i > -1; --i) {
+                        kputc(33 + qual[i], linebuf);
+                    }
+                } else {
+                    for (i = 0; i < qlen; ++i) {
+                        kputc(33 + qual[i], linebuf);
+                    }
+                }
+            }
+        } else {
+            for (i = 0; i < qlen; ++i) {
+                kputc(33 + state->def_qual, linebuf);
+            }
+        }
+        kputc('\n', linebuf);
+    }
+    return false;
+}
+
+// return false if valid
+static bool parse_opts(int argc, char *argv[], bam2fq_opts_t** opts_out)
```

Comment:
This is C, not shell.  In C, `true` means success and `false` means failure.  (That is the way around you have it in _bam_split.c_, and I was hoping you'd sort out the previous mixture in that direction…)